### ISSUE TITLE
Use filesystem level snapshots to 'freeze' the game directory before a backup

### ISSUE THREAD

**Issue by MeeniMc** [pcal43/fastback#69]
Many filesystems provide a 'snapshot' mechanism that can create local disk, well, snapshots. The snapshot is a read only view of the state of the filesystem at a prior point in time. It is cheap, as it does not need to copy blocks, it will just create new blocks for updates to the same data that is snapshot next time it is written to.

VSS for windows https://github.com/rclone/rclone/wiki/How-to-enable-VSS-for-rclone
ZFS snapshots https://ubuntu.com/tutorials/using-zfs-snapshots-clones
BTRFS https://linuxhint.com/use-btrfs-snapshots/
etc. 

This mechanism can be used to create a super cheap and rapid instant picture of the filesystem, which can then let the backup process (here git) take its sweet time in the background, meanwhile the foreground process (i.e., the minecraft server) is free to resume normal operation (including save-on activities, as needed to reduce any rollback from crashes between two snapshots).

**Comment by pcal43** [comment]
Hey, thanks for the feedback.  Right. I'm somewhat familiar with BTRFS snapshotting.

If I understand the concern correctly, I just released an update that addresses it in a somewhat simpler way.  There's now a `/backup now` command that safely backs up the world while it's running with the following sequence:
1. save the world
2. temporarily block all world saves (via a mixin)
3. update the git index with all changes (`git add`)
4. unblock world saves
5. proceed with the rest of the backup (`git commit/push`)

That step 3 only takes a few seconds in most cases I've tested, even on 5gb+ worlds - far shorter than the typical autosave window.   And step 5 is completely independent of working directory state.  So, I guess I feel like that pretty well addresses the problematic race conditions with world saves. 

And TBH I'd also be a little reluctant to dive into platform specific file system shenanigans from a Minecraft mod.

I do appreciate the thought, though.
